tests/conftest.py

```diff
@@ -972,6 +975,7 @@ def setup_func(self, module_dir: Optional[str]):
         self._keep = False
         self.project_dir = tempfile.mkdtemp()
         self.modules_dir = module_dir
+        loader.unload_modules_for_path(self.env)
```

Comment:
Shouldn't this be in `tearDownClass`? Normal `snippetcompiler` has a shared venv.

Reply:
But not every test case is using the `snippetcompiler`, so shouldn't we unload the modules after each test case (`tear_down_func`)? Some also use the `snippetcompiler_clean` or one of the `tmpvenv*` fixtures.

Reply:
Right. We could put them in both the set up and tear down for the class but I admit that's a bit cumbersome. So I agree now that this is probably the best approach.